relations/src/r1cs/constraint_system.rs

```diff
@@ -767,6 +767,16 @@ impl<F: Field> ConstraintSystemRef<F> {
         }
     }
 
+    /// Consumes self to return the inner `ConstraintSystem<F>`. Returns
+    /// `None` if `Self::CS` is `None` or if any other references to
+    /// `Self::CS` exist.  
+    pub fn into_inner(self) -> Option<ConstraintSystem<F>> {
+        match self {
+            Self::CS(a) => Some(Rc::try_unwrap(a).unwrap().into_inner()),
```

Comment:
Shouldn’t this just use a `map`?

Reply:
Updated with map in https://github.com/arkworks-rs/snark/pull/347/commits/1cfae51ff8438f77fd6876e3f7901cd357944609. Have to use a clone on `Rc` but I guess it should be okay.